internal/indexcoord/garbage_collector.go

```diff
@@ -0,0 +1,146 @@
+package indexcoord
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/milvus-io/milvus/internal/log"
+	"github.com/milvus-io/milvus/internal/storage"
+	"go.uber.org/zap"
+)
+
+type garbageCollector struct {
+	ctx    context.Context
+	cancel context.CancelFunc
+
+	wg             sync.WaitGroup
+	gcFileDuration time.Duration
+	gcMetaDuration time.Duration
+
+	metaTable    *metaTable
+	chunkManager storage.ChunkManager
+}
+
+func newGarbageCollector(ctx context.Context, meta *metaTable, chunkManager storage.ChunkManager) *garbageCollector {
+	ctx, cancel := context.WithCancel(ctx)
+	return &garbageCollector{
+		ctx:            ctx,
+		cancel:         cancel,
+		gcFileDuration: Params.IndexCoordCfg.GCInterval,
+		gcMetaDuration: time.Second * 10,
+		metaTable:      meta,
+		chunkManager:   chunkManager,
+	}
+}
+
+func (gc *garbageCollector) Start() {
+	gc.wg.Add(1)
+	go gc.recycleUnusedMeta()
+
+	gc.wg.Add(1)
+	go gc.recycleUnusedIndexFiles()
+}
+
+func (gc *garbageCollector) Stop() {
+	gc.cancel()
+	gc.wg.Wait()
+}
+
+func (gc *garbageCollector) recycleUnusedMeta() {
+	defer gc.wg.Done()
+	log.Info("IndexCoord garbageCollector recycleUnusedMetaLoop start")
+
+	ticker := time.NewTicker(gc.gcMetaDuration)
```

Comment:
call ticker.Stop after  recycleUnusedMeta returned or you may leak memory.
defer ticker.Stop

Reply:
done